kong2kic/route.go

```diff
@@ -356,12 +356,21 @@ func populateKICIngressesWithGatewayAPI(content *file.Content, kicContent *KICCo
 			})
 
 			// add service details to HTTPBackendRef
-			portNumber := k8sgwapiv1.PortNumber(*service.Port)
-			backendRef := k8sgwapiv1.BackendRef{
-				BackendObjectReference: k8sgwapiv1.BackendObjectReference{
-					Name: k8sgwapiv1.ObjectName(*service.Name),
-					Port: &portNumber,
-				},
+			var backendRef k8sgwapiv1.BackendRef
+			if service.Port != nil {
+				portNumber := k8sgwapiv1.PortNumber(*service.Port)
+				backendRef = k8sgwapiv1.BackendRef{
+					BackendObjectReference: k8sgwapiv1.BackendObjectReference{
+						Name: k8sgwapiv1.ObjectName(*service.Name),
+						Port: &portNumber,
+					},
+				}
+			} else {
+				backendRef = k8sgwapiv1.BackendRef{
+					BackendObjectReference: k8sgwapiv1.BackendObjectReference{
+						Name: k8sgwapiv1.ObjectName(*service.Name),
+					},
+				}
 			}
```

Comment:
Just to simplify:
```suggestion
			backendRef := k8sgwapiv1.BackendRef{
				BackendObjectReference: k8sgwapiv1.BackendObjectReference{
					Name: k8sgwapiv1.ObjectName(*service.Name),
				},
			}
			if service.Port != nil {
				portNumber := k8sgwapiv1.PortNumber(*service.Port)
				backendRef.Port = &portNumber
			}
```

Reply:
Thanks @czeslavo. Should line 366 be `backendRef.BackendObjectReference.Port = &portNumber`? Strangely enough, your code compiles, although I don't see any `Port` field in the `BackendRef` struct.

Reply:
It would be equivalent - it's because `BackendObjectReference` is embedded in `BackendRef` structure (https://gobyexample.com/struct-embedding).

Reply:
True! Suggestion applied in commit e5c9ed1